_posts/2020-04-20-schedule-your-functions.md

```diff
@@ -0,0 +1,130 @@
+
+Now lets roll up our sleeves and see how we can execute our OpenFaaS functions on time.
+
+## Prerequisites
+
+Before we start we need a couple of tools to help us quickly set up our environment:
+
+* [docker](https://docs.docker.com/get-docker/) - the container runtime used in this post
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start/) - lightweight local Kubernetes cluster which uses containers as nodes
+* [arkade](https://github.com/alexellis/arkade) - one line installation of applications with which we will install OpenFaaS
+* [faas-cli](https://docs.openfaas.com/cli/install/) - the CLI which communicates with the OpenFaaS gateway
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) - the CLI which communicates with your Kubernetes cluster
+
+## Prepare environment
+
+Before showing you how to run our functions on a schedule, first we need to set up the environment with the tools mentioned above.
+
+### Create cluster
+
+First thing we need is running Kubernetes cluster:
+
+```bash
+kind create cluster
+```
+
+Wait for the installation to finish and verify the pods in the `kube-system` namespace are `1/1`:
+
+```bash
+kubectl get pods -n kube-system
+```
+
+### Install OpenFaaS
+
+With `arkade` the installation of OpenFaaS boils down to single line command:
+
+```bash
+arkade install openfaas
+```
+
+Wait for the OpenFaaS gateway to be ready:
+
+```bash
+kubectl rollout status -n openfaas deploy/gateway
+```
+
+### Set up OpenFaaS client
+
+Get the OpenFaaS Gateway password:
+
+```bash
+PASSWORD=$(kubectl get secret -n openfaas basic-auth -o jsonpath="{.data.basic-auth-password}" | base64 --decode; echo)
+```
+
+Set up the gateway endpoint for the client:
+
+```bash
+export OPENFAAS_URL=http://$(docker inspect -f "{{ .NetworkSettings.IPAddress }}" $(docker ps -aqf "name=kind-control-plane")):31112
+```
+
+Authenticate the client:
+
+```bash
+echo -n $PASSWORD | faas-cli login --username admin --password-stdin
+```
+
+## Schedule the functions
+
+After we have the environment up and running, now we can install the cron-connector and schedule our functions to run at specific intervals.
+
+### Deploy the cron-connector
+
+In order to install our `cron-connector` and schedule a function run the following command:
+
+```bash
+curl -s https://raw.githubusercontent.com/openfaas-incubator/cron-connector/master/yaml/kubernetes/connector-dep.yml | kubectl apply -f -
+```
+
+Wait for the `cron-connector` to be deployed:
+
+```bash
+kubectl rollout status -n openfaas deploy/sample-connector
+```
+
+## Run the function
+
+We will run a function from the store once a minute with a `schedule` annotation following this cron syntax `*/1 * * * *` which is the same as in the [linux cron configuration](https://www.cyberciti.biz/faq/how-do-i-add-jobs-to-cron-under-linux-or-unix-oses/).
```

Comment:
"which is the same as in the [linux cron configuration]" -> See this blog post for some examples of the cron format.

Reply:
Changed as suggested :+1: